internal/wasm/httpfetcher.go

```diff
@@ -0,0 +1,224 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+
+	"github.com/envoyproxy/gateway/internal/logging"
+)
+
+// Default values for ExponentialBackOff.
+const (
+	defaultInitialInterval = 500 * time.Millisecond
+	defaultMaxInterval     = 60 * time.Second
+	maxWasmSize            = 1024 * 1024 * 256
+)
+
+var (
+	// Referred to https://en.wikipedia.org/wiki/Tar_(computing)#UStar_format
+	tarMagicNumber = []byte{0x75, 0x73, 0x74, 0x61, 0x72}
+	// Referred to https://en.wikipedia.org/wiki/Gzip#File_format
+	gzMagicNumber = []byte{0x1f, 0x8b}
+)
+
+// HTTPFetcher fetches remote wasm module with HTTP get.
+type HTTPFetcher struct {
+	client          *http.Client
+	insecureClient  *http.Client
+	initialBackoff  time.Duration
+	requestMaxRetry int
+	logger          logging.Logger
+}
+
+// NewHTTPFetcher create a new HTTP remote wasm module fetcher.
+// requestTimeout is a timeout for each HTTP/HTTPS request.
+// requestMaxRetry is # of maximum retries of HTTP/HTTPS requests.
+func NewHTTPFetcher(requestTimeout time.Duration, requestMaxRetry int, logger logging.Logger) *HTTPFetcher {
+	if requestTimeout == 0 {
+		requestTimeout = 5 * time.Second
+	}
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	// nolint: gosec
+	// This is only when a user explicitly sets a flag to enable insecure mode
+	transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
is this configurable in the UI ?

Reply:
Not yet, we can expose it to API when needed.